### ISSUE TITLE
How to get Azure User Data?

### ISSUE THREAD

**Issue by anormore** [psignoret/aad-sso-wordpress#134]
Love this plugin.

How does one pull in their Azure photo? We're building a front end, and I need their profile photo from Azure.

Maybe something like, $current_user->data->azure->display_photo ?

**Comment by psignoret** [comment]
If the thumbnail photo is in Azure AD, then this is possible, but requires a slight modification. Would you be using the photo from another plugin or a theme?

**Comment by anormore** [comment]
Thanks for the quick reply!

We want to pull ALL user data authentication from Azure AD. So this means the profile photo set in Azure is the right one. 

We can even pull that back and store locally on the WP side if need be, we just need to access it some how.

**Comment by psignoret** [comment]
Ok, and if this is made available (for example, in a variable), how would you consume it? Via a plugin or a theme?

**Comment by anormore** [comment]
Ahh I see.

Can’t this be made from your existing Azure plugin?

We would be access via theme at the moment.

Can it assign to current_user ?

**Comment by psignoret** [comment]
Yes, this plugin can retrieve the data. My question is: then what? What would you do with it once it's as retrieved and made available in some way?

**Comment by anormore** [comment]
Once the data is retrieved, can we add it to the user_meta table for storage?

I can then do $wpdb find azure_pic, if result 0, $azure->get_profile_pic from your plugin method – for example.

This allows me to display a users profile pic, from Azure.

I would probably do this action on LOGIN and REGISTER, so that if a user changes their profile pic, it will sync at next login.

**Comment by bradkovach** [comment]
At a maximum, I think we should make this available available by a hook/filter via a companion plugin but syncing anything beyond role mapping seems like a rare use case.

**Comment by richardjgreen** [comment]
This would be awesome for a project I am currently looking at. Instead having each author on the site registering with Gravatar and having WordPress display a Gravatar image for the author, we could pull the profile image for the user from Azure AD.

The users of my project will not be the most tech savvy, so making it as seamless and easy for them to get it right would be ideal.

**Comment by jjacobs1** [comment]
We also would love this feature

**Comment by soonerdude** [comment]
I'd like to be able to use Wordpress as an Intranet site for our company. We use Office 365, so it would be great if our users can authenticated using this plugin, and then have their common information downloaded / mapped to some custom wordpress fields, such as Department, Title (jobTitle in MS graph), Phone, and email address. Then it should be fairly easy to create an employee directory that can be filtered by department / title, etc.

**Comment by psignoret** [comment]
@anormore Can you share the details of the approach you're taking? I've almost finished this already, so we should avoid double work here.

**Comment by brandontravis** [comment]
Hello @psignoret and @anormore ... has any progress been made here?

**Comment by ferd-i** [comment]
+1

**Comment by milardovich** [comment]
+1
